src/tools.ts

```diff
@@ -27,420 +38,364 @@ class ToolException extends Error {
 
 /**
  * Process the result from calling an MCP tool.
+ * Extracts text content when available for better agent compatibility.
  *
  * @param result - The result from the MCP tool call
  * @returns The processed result
  */
-function _convertCallToolResult(result: CallToolResult): any {
+function _convertCallToolResult(result: CallToolResult): unknown {
   logger.debug('Processing MCP tool result:', JSON.stringify(result, null, 2));
```

Comment:
is there a way to disable logging? some verbosity parameter people can set? i'd prefer that it was disabled by default and was an opt-in behavior